File: src/main/java/net/gtaun/shoebill/streamer/data/DynamicRectangle.java
package net.gtaun.shoebill.streamer.data;

import net.gtaun.shoebill.data.Area;
import net.gtaun.shoebill.streamer.Functions;

/**
 * Created by Valeriy on 18/4/2016.
 */
public class DynamicRectangle extends DynamicArea {
    public DynamicRectangle(int id, int playerId) {
        super(id, playerId);
    }

    public static DynamicRectangle create(Area area, int worldId, int interiorId, int playerId) {
        DynamicRectangle rectangle = Functions.createDynamicRectangle(area, worldId, interiorId, playerId);
        DynamicArea.addArea(rectangle);
        return rectangle;
    }
}

File: src/main/java/net/gtaun/shoebill/streamer/data/DynamicRectangle.kt
package net.gtaun.shoebill.streamer.data

import net.gtaun.shoebill.entities.Player
import net.gtaun.shoebill.data.Area
import net.gtaun.shoebill.streamer.AllOpen
import net.gtaun.shoebill.streamer.Functions

/**
 * @author Valeriy
 */
@AllOpen
class DynamicRectangle(id: Int, player: Player?) : DynamicArea(id, player) {
    companion object {
        @JvmOverloads
        @JvmStatic
        fun create(area: Area, worldId: Int = -1, interiorId: Int = -1, player: Player? = null): DynamicRectangle {
            val playerId = player?.id ?: -1

            val rectangle = Functions.createDynamicRectangle(area, worldId, interiorId, playerId)
            areas.add(rectangle)
            return rectangle
        }
    }
}
